### PR TITLE
New job

### PR DESCRIPTION
**Pull request by rozyczko**
Virtual classes for the new job and relevant other things.

Also, changes to the lmfit wrapper fixing circular dependencies.

### CODE REVIEW

File: src/easyscience/Objects/Job/TheoreticalModel.py [hunk 1]
```diff
@@ -6,19 +6,19 @@
...
 
-class TheoryBase(BaseObj):
+class TheoreticalModelBase(BaseObj):
     """
     This virtual class allows for the creation of technique-specific Theory objects.
     """
     def __init__(self, name: str, *args, **kwargs):
         self._name = name
-        super(TheoryBase, self).__init__(name, *args, **kwargs)
+        super(TheoreticalModelBase, self).__init__(name, *args, **kwargs)
```
Comment (andped10) [inline src/easyscience/Objects/Job/TheoreticalModel.py]: I think you could just use: `super().__init__(name, *args, **kwargs)`

File: src/easyscience/Objects/Job/TheoreticalModel.py [hunk 2]
```diff
@@ -6,19 +6,19 @@
...
 
     # required dunder methods
     def __str__(self):
         raise NotImplementedError("Copy not implemented")
     
     def as_dict(self, skip: list = []) -> dict:
-        this_dict = super(TheoryBase, self).as_dict(skip=skip)
+        this_dict = super(TheoreticalModelBase, self).as_dict(skip=skip)
```
Comment (andped10) [inline src/easyscience/Objects/Job/TheoreticalModel.py]: Similar to above: `this_dict = super().as_dict(skip=skip)`

File: src/easyscience/Objects/Job/TheoreticalModel.py
Comment (andped10) [inline src/easyscience/Objects/Job/TheoreticalModel.py:1]: Change to small caps filename

File: src/easyscience/Objects/Job/Job.py
Comment (andped10) [inline src/easyscience/Objects/Job/Job.py:1]: Change to small caps filename

File: src/easyscience/Objects/Job/Job.py [hunk 3]
```diff
@@ -74,10 +74,10 @@ def analysis(self, analysis: AnalysisBase):
     # def info(self, info: InfoBase):
     #     raise NotImplementedError("info setter not implemented")
 
-    @abstractmethod
-    def calculate_theory(self, *args, **kwargs):
-        raise NotImplementedError("calculate_theory not implemented")
-        #pass
+    # @abstractmethod
```
Comment (andped10) [inline src/easyscience/Objects/Job/Job.py]: Should this be deleted?

File: src/easyscience/Objects/Job/Analysis.py
Comment (andped10) [inline src/easyscience/Objects/Job/Analysis.py:1]: Change to small caps filename

File: src/easyscience/Objects/Job/Analysis.py [hunk 4]
```diff
@@ -1,18 +1,60 @@
 #  SPDX-FileCopyrightText: 2023 EasyScience contributors  <core@easyscience.software>
 #  SPDX-License-Identifier: BSD-3-Clause
 #  © 2021-2023 Contributors to the EasyScience project <https://github.com/easyScience/EasyScience
+from abc import ABCMeta
+from abc import abstractmethod
+from typing import Union
 
+import numpy as np
 
+from easyscience.Datasets.xarray import xr  # type: ignore
```
Comment (andped10) [inline src/easyscience/Objects/Job/Analysis.py]: Are we not moving away from xarray?

File: src/easyscience/Objects/Job/Analysis.py [hunk 5]
```diff
@@ -1,18 +1,60 @@
...
+        raise NotImplementedError("fit not implemented")
+
+    @property
+    def calculator(self):
+        if self._calculator is None:
+            self._calculator = self.interface.current_interface_name
+        return self._calculator
+
+    @calculator.setter
+    def calculator(self, value):
```
Comment (andped10) [inline src/easyscience/Objects/Job/Analysis.py]: please add type

File: src/easyscience/Objects/Job/Analysis.py [hunk 6]
```diff
@@ -1,18 +1,60 @@
...
+        # TODO: check if the calculator is available for the given JobType
+        self.interface.switch(value, fitter=self._fitter)
+
+    @property
+    def minimizer(self):
+        return self._minimizer
+
+    @minimizer.setter
+    def minimizer(self, minimizer):
```
Comment (andped10) [inline src/easyscience/Objects/Job/Analysis.py]: please add type

**Review by andped10 — APPROVED** [review]

File: src/easyscience/Objects/Job/Job.py (hunk 3 above)
Comment (rozyczko) [reply src/easyscience/Objects/Job/Job.py]: The summary/info abstract classes will be added later, so I left the code for future reuse.
The `calculate_theory` abstract method should not be commented out. Fixed.

File: src/easyscience/Objects/Job/Analysis.py (hunk 4 above)
Comment (rozyczko) [reply src/easyscience/Objects/Job/Analysis.py]: We are, but conversion to scipp in libraries will be done gradually. I could remove the dependency on xarray in EasyScience, though, by deleting type hints.

**Pull request merged**